package/network/utils/comgt/files/ncm.sh

```diff
@@ -117,21 +127,23 @@ proto_ncm_setup() {
 	}
 
 	echo "Starting network $interface"
-	json_get_vars connect
-	eval COMMAND="$connect" gcom -d "$device" -s /etc/gcom/runcommand.gcom || {
-		echo "Failed to connect"
-		proto_notify_error "$interface" CONNECT_FAILED
-		return 1
-	}
-
 	echo "Setting up $ifname"
-	
+
 	proto_init_update "$ifname" 1
 	proto_add_data
 	json_add_string "manufacturer" "$manufacturer"
 	proto_close_data
 	proto_send_update "$interface"
 
+	json_load "$(cat /etc/gcom/ncm.json)"
+	json_select "$manufacturer"
+	json_get_vars connect
+	eval COMMAND="$connect" gcom -d "$device" -s /etc/gcom/runcommand.gcom || {
+		echo "Failed to connect"
+		proto_notify_error "$interface" CONNECT_FAILED
+		return 1
+	}
+
```

Comment:
Hi Matti,
I cannot understand why the error do not occurs when I use the modified ncm.sh (the one delaying the AT+CGDATA="M-MBIM",1,1 command) but only with the original one.
The runcommand.gcom is actually the same.

Bye